hypothesis-python/src/hypothesis/internal/conjecture/shrinker.py

```diff
@@ -675,22 +668,18 @@ def greedy_shrink(self):
         """
         self.fixate_shrink_passes(
             [
-                block_program("X" * 5),
-                block_program("X" * 4),
-                block_program("X" * 3),
-                block_program("X" * 2),
-                block_program("X" * 1),
+                node_program("X" * 5),
+                node_program("X" * 4),
+                node_program("X" * 3),
+                node_program("X" * 2),
+                node_program("X" * 1),
                 "pass_to_descendant",
                 "reorder_examples",
-                "minimize_floats",
-                "minimize_duplicated_blocks",
-                block_program("-XX"),
-                "minimize_individual_blocks",
-                block_program("--X"),
+                "minimize_duplicated_nodes",
+                "minimize_individual_nodes",
```

Comment:
one would hope not! if you can think of a case where block level logic is important, please do tell me, because that's a threat to validity of the ir project as a whole. Blocks won't exist for much longer after all.

I think the only regressions we'd see here — and we are indeed seeing them — is places where the block program has a lucky side effect that causes e.g. a shrink across type boundaries, which is harder in the ir. Not that it can't be done, but it requires more engineering.

Reply:
Ah, I'm confusing Blocks and Examples here; never mind.